vcr/signature/ldutils.go

```diff
@@ -42,6 +57,21 @@ func NewEmbeddedFSDocumentLoader(fs embed.FS, nextLoader ld.DocumentLoader) ld.D
 	}
 }
 
+// NewFilteredLoader accepts a list of allowed urls and a nextLoader and creates a new filteredDocumentLoader
+func NewFilteredLoader(allowedURLs []string, nextLoader ld.DocumentLoader) ld.DocumentLoader {
+	return &filteredDocumentLoader{AllowedURLs: allowedURLs, nextLoader: nextLoader}
+}
+
+// LoadDocument calls the nextLoader if the url u is on the AllowedURLs list, throws a ld.LoadingDocumentFailed otherwise.
+func (h filteredDocumentLoader) LoadDocument(u string) (*ld.RemoteDocument, error) {
+	for _, allowedURL := range h.AllowedURLs {
+		if allowedURL == u {
+			return h.nextLoader.LoadDocument(u)
+		}
+	}
+	return nil, ld.NewJsonLdError(ld.LoadingDocumentFailed, nil)
+}
+
 // LoadDocument tries to load the document from the embedded filesystem.
 // If the document is not a file or could not be found it tries the nextLoader.
 func (e embeddedFSDocumentLoader) LoadDocument(path string) (*ld.RemoteDocument, error) {
```

Comment:
just checked the inners of this function, why does it check the URI scheme for being http or https if it only reads from the embedded fs?

And also given;

```
		file, err := e.fs.Open(path)
		if err != nil {
			return e.nextLoader.LoadDocument(path)
		}
```
Doesn't this highly depend on the error that occured? You probably only want to go to the next loader if the file didn't exist (which is easily checkable), and otherwise fail?

Reply:
The loaders can all handle their own type. This one handles files and ignores http(s) based resources, hence the http(s) check. If the file is not found on the embedded FS, pass it on to the nextLoader.